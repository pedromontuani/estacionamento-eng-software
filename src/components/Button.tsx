import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

interface IButtonProps {
  title: string;
  onPress(): void;
  testID?: string;
}

const Button: React.FC<IButtonProps> = ({testID, title, onPress}) => {
  return (
    <TouchableOpacity
      testID={testID}
      style={styles.container}
      onPress={onPress}>
      <Text style={styles.text}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 30,
    width: '100%',
    borderRadius: 20,
    backgroundColor: '#1976d2',
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 5,
  },
  text: {
    fontSize: 14,
    color: 'white',
    fontWeight: '500',
  },
});

export default Button;
