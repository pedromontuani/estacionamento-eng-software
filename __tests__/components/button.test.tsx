import React from 'react';
import {rendererCreate} from '../utils';

import Button from '../../src/components/Button';

describe('Testing Button component', () => {
  test('should render correctly', async () => {
    const tree = rendererCreate(<Button title="Teste" onPress={() => {}} />);
    expect(tree).toMatchSnapshot();
  });
});
