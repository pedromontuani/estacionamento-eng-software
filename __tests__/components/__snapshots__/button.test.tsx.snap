// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1976d2",
      "borderRadius": 20,
      "elevation": 5,
      "height": 30,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 14,
        "fontWeight": "500",
      }
    }
  >
    Teste
  </Text>
</View>
`;
